### ISSUE TITLE
Why Did Mozilla Remove XUL Add-ons?

### ISSUE THREAD

**Issue by Yoric** [Yoric/yoric.github.io#2]
https://yoric.github.io/post/why-did-mozilla-remove-xul-addons/ 

 TL;DR: Firefox used to have a great extension mechanism based on the XUL and XPCOM. This mechanism served us well for a long time. However, it came at an ever-growing cost in terms of maintenance for both Firefox developers and add-on developers. On one side, this growing cost progressively killed any effort to make Firefox secure, fast or to try new things. On the other side, this growing cost progressively killed the community of add-on developers. Eventually, after spending years trying to protect this old add-on mechanism, Mozilla made the hard choice of removing this extension mechanism and replacing this with the less powerful but much more maintainable WebExtensions API. Thanks to this choice, Firefox developers can once again make the necessary changes to improve security, stability or speed.
During the past few days, I&rsquo;ve been chatting with Firefox users, trying to separate fact from rumor regarding the consequences of the August 2020 Mozilla layoffs. One of the topics that came back a few times was the removal of XUL-based add-ons during the move to Firefox Quantum. I was very surprised to see that, years after it happened, some community members still felt hurt by this choice.
And then, as someone pointed out on reddit, I realized that we still haven&rsquo;t taken the time to explain in-depth why we had no choice but to remove XUL-based add-ons.
So, if you&rsquo;re ready for a dive into some of the internals of add-ons and Gecko, I&rsquo;d like to take this opportunity to try and give you a bit more detail.
